enterprise/server/hit_tracker_client/hit_tracker_client.go

```diff
@@ -164,16 +171,27 @@ type HitTrackerClient struct {
 	enqueueFn       func(context.Context, *repb.RequestMetadata, *hitpb.CacheHit)
 	client          hitpb.HitTrackerServiceClient
 	requestMetadata *repb.RequestMetadata
+	cacheType       rspb.CacheType
 }
 
-// This should only be used for Action Cache hit tracking.
 func (h *HitTrackerClient) SetExecutedActionMetadata(md *repb.ExecutedActionMetadata) {
+	// This is used to track action durations and is not used for non-RBE executions.
```

Comment:
Someone (You? Jim?) said that this should only be called with nil in the case you're adding, so I'd just check if it's nil and log an UnexpectedEvent if not here.

Reply:
For workflows, the hit tracker does not need the action metadata, so it is safe to ignore. But the [metadata will not be nil ](https://github.com/buildbuddy-io/buildbuddy/blob/master/enterprise/server/remote_execution/snaploader/snaploader.go#L633), so we shouldn't expect it to be